### DISCUSSION TITLE
Sign-Verify, Encrypt-Decrypt and onion addresses

### DISCUSSION THREAD

**Discussion by huey735** [bisq-network/bisq#5174] [Ideas]
Sign-Verify, Encrypt-Decrypt are great functions provided by the cryptography used by bitcoin that allow users to prove custody of private keys associated with a certain public key/address.

Recently the [Mutisig Payout tool](https://github.com/bisq-network/bisq/pull/4899) was greatly improved by @jmacxx  to allow for the trust-less build, verification and broadcast of transactions. This brought Sign-Verify closer to the traders. Imagine how great it could be if traders could build-verify-sign-broadcast more type of transactions, with different payout schemes. This could greatly reduce the job of Mediators.

What if traders could Sign-Verify messages? They could prove the custody of the keys to others very easily without having to resort to other software like Electrum and risk copy pasting private keys from one application to another. The same for Encrypt-Decrypt.

At present Bisq has Trader, Mediation and Arbitration chats. Despite the huge work done on it by the developers it's very bare bones. With the tools above, the traders could still open a Mediation Ticket which would notify the Mediator but the conversation could be had on encrypted-chat apps like Keybase.

There the user would have an account named after their onion address and the Mediator could simply open a three-way chat with both traders. Still leaving the possibility for each of the three to break out into two-chats if they so intended.

This would allow the institution of practices like Support Agents introducing themselves to traders with a verifiable signed message. With Schnorr it could allow for any trader to post messages with the anonymity set of of all known public keys of Bisq trades.

**Comment by Conza88** [comment]
> At present Bisq has Trader, Mediation and Arbitration chats. Despite the huge work done on it by the developers it's very bare bones. With the tools above, the traders could still open a Mediation Ticket which would notify the Mediator but the conversation could be had on encrypted-chat apps like Keybase.

Dang. Certainly one way to bring some 'firepower' to chat etc. 

Keybase? Digg it but privacy orientated & long term issues exist with tying into it, given Zoom owned now? See variety of comments lamenting this and asking about what is next etc, open source?

I think if any UX is simple & seamless could work. Otherwise, opens up a break point?

**Reply by huey735** [reply]
As long as it's open-source it shouldn't be a big issue.

And I imagine it could be easy to simply use another end-to-end-encrypted-app.

**Reply by Conza88** [reply]
Generally speaking you want to keep folks on platform (website, app) etc. And not send them away. 
Pro's and con's - obviously if there's a rich integration then it could make sense.
I'd certainly be interested to see what it'd look like in practice. 

Each user having to create an account with a same onion address manually in Keybase, if they want to communicate with other traders/mediators/agents? :\ 

Definitely unlikely. Unless it's more seamless than the current chat and integrated, I don't see it as an improvement TBH.

**Reply by huey735** [reply]
It makes sense in my mind to have a dedicated account for dealing with Bisq Mediation. We could create a bisq.mediation team on Keybase where it'd be all onion addresses for usernames. I don't think it'd be too much of a hurdle, but I may be wrong.

There aren't any tools for making the protocol above seamless but Keybase's chat is way better than Bisq's - notifications, no p2p network issues, easy file sharing. And some times the trader end up having to reach mediator through Keybase anyway. I don't think Keybase makes it easy to have multiple accounts simultaneously open so that would add friction.

What do you guys think @jmacxx @chimp1984 @sqrrm @ripcurlx @wiz @leo816 ?

**Comment by chimp1984** [comment]
Please note that all messages in the Bisq P2P network are signed and encrypted, so the receiver is the only one who can decrypt it and verifies the senders signature. I don't think we should extend too much in 3rd party tools like Keybase. They can shut down Bisq at any moment and I do not trust any tool which is not open source over the full tech chain (I doubt Keybase servers can be inspected/audited).

**Reply by huey735** [reply]
Bisq Mediators wouldn't be locked to Keybase. As long as users can sign strings and verify signatures on Bisq, the chatting can be done elsewhere where the chatting tools are better suited for:
- notifying of new messages
- sending files
- reliability of messages being sent

Bisq chat tools can be a burden on the P2P network to transmit and store the data, developers for continued maintenance and development and those engaging in conversation due to the pain points mentioned above.

**Comment by Conza88** [comment]
What about https://twitter.com/sphinx_chat ? Chat over lightning 😅🤷‍♂️

**Comment by ghost** [comment]
Consider that not all users can install Keybase.  Like @ThisMightHelp : https://github.com/bisq-network/bisq/issues/5102#issuecomment-765672745  
I remember my own decision to install Keybase, it was not an easy hurdle to get over because I have trust issues installing new software.
You would then have to support two chat options for Mediation, existing in-app **and** Keybase.

I would also note that the Keybase chat app is quite memory hungry (as is Bisq).  On my laptop (4Gb) I cannot run them both and a browser at the same time.

**Reply by huey735** [reply]
@jmacxx I'm proposing to eventually get rid of Bisq's chat tools and have the software provide only tools to prove and verify custody of private keys.
Years ago, [when discussing the migration from Slack](https://github.com/bisq-network/proposals/issues/54#issuecomment-435684993) tox.chat was suggested by @vycoreb and rejected by @cbeams , It is in fact ill suited for all we do with Keybase but it's good enough (and better than Bisq) for Mediation.

**Comment by ghost** [comment]
@huey735 I made this sign/verify functionality which is comparable to Electrum.  Is this what you want?

Screenshot: Bisq sign/verify & Electrum sign/verify.

![image](https://user-images.githubusercontent.com/47253594/111938391-d6995580-8a97-11eb-83ce-8f2bcaac6259.png)

**Reply by huey735** [reply]
Exactly.
